src/api/app/views/webui2/webui/staging/workflows/_problems.html.haml

```diff
@@ -2,20 +2,31 @@
   max_problems = 5
   project_class = staging_project.name.tr(':', '_')
   item_project_class = 'hidden-item-' + project_class
+  left_problems = max_problems
+  failed_status_checks = staging_project.failed_status_checks
+  build_problems = staging_project.problems
+  total_length = failed_status_checks.length + build_problems.length
 
-- if staging_project.problems.empty?
+- if build_problems.empty? && failed_status_checks.empty?
   .text-center
     %i.fas.fa-check-circle.text-success
 - else
   %ul.list-group.list-group-flush
-    = render partial: 'problems_item', locals: { staging_project: staging_project, index: 0...max_problems }
-
-    - length = staging_project.problems.length
-    - if length > max_problems
+    - failed_status_checks[0...max_problems].each do |check|
+      = render partial: 'status_problems', locals: { check: check }
+      - failed_status_checks.shift
```

Comment:
why do you need this? :unamused:

Reply:
@ana06 i guess you are talking about the ```shift```, i first tried it without but it ended up using helper variables to indicate where to continue the iteration in the hidden part of the list, so i thought it easier to just remove the rendered items and just rendering the left overs in the collapsed part

Reply:
you can do this with the length, this variable is confusing and not needed